photon-core/src/main/java/org/photonvision/vision/processes/VisionSourceManager.java

```diff
@@ -62,546 +65,286 @@ public static VisionSourceManager getInstance() {
         return SingletonHolder.INSTANCE;
     }
 
-    VisionSourceManager() {}
+    public VisionModuleManager vmm = new VisionModuleManager();
 
-    public void registerTimedTask() {
-        TimedTaskManager.getInstance().addTask("VisionSourceManager", this::tryMatchCams, 3000);
-    }
+    // TODO - delete these. and replace ConcurrentHashMap with syncronization
+    private final ConcurrentHashMap<String, CameraConfiguration> deserializedConfigs =
+            new ConcurrentHashMap<>();
+    private final ConcurrentHashMap<String, PVCameraInfo> cameraDeviceMap = new ConcurrentHashMap<>();
```

Comment:
It feels fishy to have multiple bits of state but have them individually syncronized. we should use the syncronized keyword or a mutex instead. I vote for syncronized(this).